src/test/php/org/bovigo/vfs/vfsStreamWrapperFileTestCase.php

```diff
@@ -454,4 +454,22 @@ public function cannotReadFileFromNonReadableDir()
         $this->bar->chmod(0);
         $this->assertFalse(@file_get_contents($this->baz1URL));
     }
+
+    /**
+     * @test
+     * @see   https://github.com/mikey179/vfsStream/issues/155
+     * @since 1.6.5
+     */
+    public function errorOnOpen()
+    {
+        $file = $this->baz1;
+        $path = $file->url();
+        $this->assertNotFalse(@fopen($path, 'r'), 'precondition');
+
+        $file->setOpenError("fopen() gives error");
+        $this->assertFalse(@fopen($path, 'r'), 'fopen fails now');
+
+        $file->setOpenError(null);
+        $this->assertNotFalse(@fopen($path, 'r'), 'fopen works again');
+    }
 }
```

Comment:
I can change that, but I think this is not a correct rule or most often misunderstood. For the test written if it fails, I have left a message on the different expectations so that you exactly know that. The overall test-method asserts the only one thing of that feature, which is more what "one assertion per test" means. These are easy to do btw, extract a private method that starts with assert in it's name and then implement that single one assertion in the private method. That makes this more visible. So the rule has its grounds and is a good one, but when I read such comments I think that programmers sometimes read that too literally and they think there could not be more than an assertion call. And that's not what the rule is for. Hope this leaves some food for thought :) - When refactoring out the new sub-type, this is automatically different again, and I'm totally with you that such a rule is important for a good test-case.